src/core/InstrumentFunctions.cpp

```diff
@@ -435,26 +472,27 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 		// process according to arpeggio-direction...
 		if (dir == ArpDirection::Up || dir == ArpDirection::Down)
 		{
-			cur_arp_idx = ( cur_frame / arp_frames ) % range;
+			cur_arp_idx = (cur_frame / arp_frames) % (range + sortOffset);
```

Comment:
`range + sortOffset` is the same as `total_range`. I think you can skip `sortOffset` alltogether.

Reply:
It is only total_range when sorting.

Reply:
Ah, got it!